data/shaders/opengl/geosphere_terrain.frag

```diff
@@ -49,21 +54,31 @@ void main(void)
 	vec4 detailMul = mix(vec4(1.0), detailVal, detailMix);
 
 #ifdef TERRAIN_WITH_WATER
-	float specularReflection=0.0;
+	vec4 waterSpecular = vec4(0.0);
 #endif
 
 #if (NUM_LIGHTS > 0)
 	vec3 V = normalize(eyeposScaled - geosphereCenter);
+	float AU = 149598000000.0;
+
+	vec3 I = normalize(eyeposScaled - planetIntersect);
+	vec3 center = I * geosphereRadius;
+	frag_color = vec4(0.f);
 
 	for (int i=0; i<NUM_LIGHTS; ++i) {
 		vec3 L = normalize(uLight[i].position.xyz);
+
 		float uneclipsed = clamp(calcUneclipsed(eclipse, NumShadows, V, L), 0.0, 1.0);
 		CalcPlanetDiffuse(diff, uLight[i].diffuse, L, tnorm, uneclipsed);
 
 #ifdef TERRAIN_WITH_WATER
 		//water only for specular
-	    if (vertexColor.b > 0.05 && vertexColor.r < 0.05) {
-			CalcPlanetSpec(specularReflection, uLight[i], L, tnorm, eyenorm, WATER_SHINE);
+		if (vertexColor.b > 0.05 && vertexColor.r < 0.05) {
+			// Convert from radius-relative to real coordinates
+			vec3 lightPosAU = uLight[i].position.xyz / AU;
```

Comment:
if AU is const, then you can have a constant called `INV_AU = 1.0f / AU`. Then change this line to something like `uLight[i].position.xyz * INV_AU;` as mul is faster than div